kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/TopicPatternSubscriber.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.axonframework.extensions.kafka.eventhandling.consumer;
+
+import org.apache.kafka.clients.consumer.Consumer;
+
+import java.util.regex.Pattern;
+
+/**
+ * Implementation of {@link TopicSubscriber} that subscribes a {@link Consumer} to a pattern of topics.
+ * Using the {@link Consumer#subscribe(Pattern)} method.
+ *
+ * @author Ben Kornmeier
+ * @since 4.8.0
+ */
+public class TopicPatternSubscriber implements TopicSubscriber {
+    private final Pattern pattern;
+
+    /**
+     * Instantiate a {@link TopicPatternSubscriber} that uses {@link Pattern} to subscribe to topics as well as check if it is responsible for a given topic.
+     *
+     * @param pattern
+     */
+    public TopicPatternSubscriber(Pattern pattern) {
+        this.pattern = pattern;
+    }
+
+    /**
+     * Subscribes the given {@link Consumer} to the topic(s) using the {@link Consumer#subscribe(Pattern)} method.
```

Comment:
Nit: although it's not wrong, implemented/overridden methods receive the JavaDoc from the interface/parent class. Hence, this is a duplication.